Firmware/stepper.cpp

```diff
@@ -1301,62 +1301,67 @@ void st_init()
     disable_e2();
   #endif
 
-  // waveform generation = 0100 = CTC
-  TCCR1B &= ~(1<<WGM13);
-  TCCR1B |=  (1<<WGM12);
-  TCCR1A &= ~(1<<WGM11);
-  TCCR1A &= ~(1<<WGM10);
-
-  // output mode = 00 (disconnected)
-  TCCR1A &= ~(3<<COM1A0);
-  TCCR1A &= ~(3<<COM1B0);
-
-  // Set the timer pre-scaler
-  // Generally we use a divider of 8, resulting in a 2MHz timer
-  // frequency on a 16MHz MCU. If you are going to change this, be
-  // sure to regenerate speed_lookuptable.h with
-  // create_speed_lookuptable.py
-  TCCR1B = (TCCR1B & ~(0x07<<CS10)) | (2<<CS10);
-
-  // Plan the first interrupt after 8ms from now.
-  OCR1A = 0x4000;
-  TCNT1 = 0;
+  ATOMIC_BLOCK(ATOMIC_RESTORESTATE)
+  {
+    // waveform generation = 0100 = CTC
+    TCCR1B &= ~(1<<WGM13);
+    TCCR1B |=  (1<<WGM12);
+    TCCR1A &= ~(1<<WGM11);
+    TCCR1A &= ~(1<<WGM10);
+
+    // output mode = 00 (disconnected)
+    TCCR1A &= ~(3<<COM1A0);
+    TCCR1A &= ~(3<<COM1B0);
+
+    // Set the timer pre-scaler
+    // Generally we use a divider of 8, resulting in a 2MHz timer
+    // frequency on a 16MHz MCU. If you are going to change this, be
+    // sure to regenerate speed_lookuptable.h with
+    // create_speed_lookuptable.py
+    TCCR1B = (TCCR1B & ~(0x07<<CS10)) | (2<<CS10);
+
+    // Plan the first interrupt after 8ms from now.
+    OCR1A = 0x4000;
+    TCNT1 = 0;
 
```

Comment:
I need to re-verify this, but we should be able to close the atomic brace here, like we do in st_reset_timer.

Reply:
It scares me that planner variables are still initialized after the logic analyzer stuff. Doesn't seem healthy to start the stepper ISR with these values not initialized.

Reply:
I would move the whole LIN_ADVANCE block just before the timer initialization then, so that we can keep the atomic block just for the timer setup.

Reply:
Ok. will do that. Just not right now